sw/device/lib/dif/dif_usbdev.h

```diff
@@ -199,7 +199,18 @@ dif_result_t dif_usbdev_endpoint_setup_enable(const dif_usbdev_t *usbdev,
                                               dif_toggle_t new_state);
 
 /**
- * Enable or disable reception of OUT packets for an endpoint.
+ * Enable or disable reception of OUT packets for an active endpoint.
+ *
+ * When disabling reception of OUT packets, what the endpoint will do depends
+ * on other factors. If the endpoint is currently configured as a control
+ * endpoint (receives SETUP packets) or it is configured as an isochronous
+ * endpoint, disabling reception of OUT packets will cause them to be ignored.
+ *
+ * If the endpoint is neither a control nor isochronous endpoint, then its
+ * behavior depends on whether it is configured to respond with STALL. If the
+ * STALL response is not active, then disabling reception will cause usbdev to
+ * NAK the packet. Otherwise, the STALL response takes priority, regardless of
+ * the setting here.
  *
  * @param usbdev A USB device.
  * @param endpoint An OUT endpoint number.
```

Comment:
nit:
```suggestion
 * @param[out] endpoint An OUT endpoint number.
```

Reply:
(applies throughout)

Reply:
Uh, that's not correct. It is not an output parameter. The "OUT" here refers to USB transaction directions.

Reply:
oops sorry you are right, the way it looked on my screen i thought this comment applied to  `dif_result_t dif_usbdev_address_get()`... the code blocked was collapsed. My apologies!

Reply:
Hm... But it does look like there are some issues for the older code.

Reply:
SG. yeah I saw that but didn't think it was fair to call out as that was not your change. But if you feel inclined to fixed, by all means hehe. Thanks @a-will !